src/View/mainpage/DrawROIWindow/Drawing.py

```diff
@@ -115,15 +117,44 @@ def _display_pixel_color(self):
                 colors = QColor(c).getRgbF()
                 self.according_color_dict[(x_coord, y_coord)] = colors
 
-            color = QtGui.QColor()
-            color.setRgb(90, 250, 175, 200)
-            points = get_pixel_coords(self.according_color_dict, self.rows,
-                                      self.cols)
-            for x_coord, y_coord in points:
-                self.q_image.setPixelColor(x_coord, y_coord, color)
+            points = get_pixel_coords(self.according_color_dict, self.rows, self.cols)
+
+            self._set_color_of_pixels(points)
 
             self.refresh_image()
 
+    def _set_color_of_pixels(self, points):
+        """
+        Takes a set of points that need to be coloured, and updates the colour of
+        each pixel with the drawn on colour, blended together based on the transparency value.
+
+        :param points:  Set of x and y co-ordinates of pixels to be coloured
+        """
+        logging.debug("_set_color_of_pixels started")
+        img = self.img.toImage()
+        for x_coord, y_coord in points:
+            old_color = img.pixelColor(x_coord, y_coord)
+            # Blend the individual RGB values of the original pixel with the drawn ROI colour values
+            new_color = QtGui.QColor(self.drawn_pixel_color.red() * (1 - self.pixel_transparency)
+                                     + old_color.red() * self.pixel_transparency,
+                                     self.drawn_pixel_color.blue() * (1 - self.pixel_transparency)
+                                     + old_color.blue() * self.pixel_transparency,
+                                     self.drawn_pixel_color.green() * (1 - self.pixel_transparency)
+                                     + old_color.green() * self.pixel_transparency,
+                                     255)
+            self.q_image.setPixelColor(x_coord, y_coord, new_color)
+        logging.debug("_set_color_of_pixels finished")
+
+    def update_pixel_transparency(self):
+        """
+        Updates the transparency of pixels that have already been drawn, with new
+        transparency value set by user and refreshes the image.
+        """
+        logging.debug("update_pixel_transparency started")
+        self._set_color_of_pixels(get_pixel_coords(self.target_pixel_coords, self.rows, self.cols))
+        self.refresh_image()
+        logging.debug("update_pixel_transparency finished")
```

Comment:
Given what was discussed in the meeting should I keep these logging statements in? Possibly comment out?

Reply:
Debug statements will stay in